src/TodosContext/TodosContext.tsx

```diff
@@ -0,0 +1,390 @@
+  const [temptTodo, setTempTodo] = useState<Todo | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filteringBy, setFilteringBy] = useState(FilteringBy.default);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [startValue, setStartValue] = useState('');
+  const [processingIds, setProcessingIds] = useState<number[]>([]);
+
+  useEffect(() => {
+    todoService
+      .getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToLoadTodos);
+
+        setTimeout(() => setErrorMessage(null), 3000);
+      });
+  }, []);
+
+  const isAllCompleted = useMemo(() => {
+    return todos.every(todo => todo.completed);
+  }, [todos]);
+
+  const handleSpanDoubleClick = (id: number, title: string) => {
+    setEditedTodoId(id);
+    setEditedTodoTitle(title);
+    setStartValue(title);
+  };
+
+  const handleDeleteEmptyTodo = (todoId: number) => {
+    setIsSubmiting(true);
+    setEditedTodoId(0);
+    setProcessingIds(prev => [...prev, todoId]);
+    todoService
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos(prev => prev.filter(todo => todo.id !== todoId));
+        setIsSubmiting(false);
+        setProcessingIds(prev =>
+          prev.filter(currTodoId => currTodoId !== todoId),
+        );
+      })
+      .catch(() => {
+        setIsSubmiting(false);
+        setErrorMessage(ErrorMessage.UnableToDeleteATodo);
+        setEditedTodoId(todoId);
+        setProcessingIds(prev =>
+          prev.filter(currTodoId => currTodoId !== todoId),
+        );
+        setTimeout(() => setErrorMessage(null), 3000);
+      });
+  };
+
+  const handleEditTodo = (
+    event: React.FormEvent<HTMLFormElement>,
+    todoId: number,
+  ) => {
+    event.preventDefault();
+
+    if (editedTodoTitle) {
+      if (startValue !== editedTodoTitle) {
+        setEditedTodoId(null);
+        setProcessingIds(curr => [...curr, todoId]);
+        todoService
+          .editTodo(todoId, { title: editedTodoTitle })
+          .then(() => {
+            setTodos(prev =>
+              prev.map(todo => {
+                if (todo.id === todoId) {
+                  return { ...todo, title: editedTodoTitle.trim() };
+                }
+
+                return todo;
+              }),
+            );
+            setProcessingIds(prev =>
+              prev.filter(currentTodo => currentTodo !== todoId),
+            );
+          })
+          .catch(() => {
+            setErrorMessage(ErrorMessage.UnableToUpdateATodo);
+            setTimeout(() => setErrorMessage(null), 3000);
+            setEditedTodoId(todoId);
+            setProcessingIds(prev =>
+              prev.filter(currTodo => currTodo !== todoId),
+            );
+          });
+      } else {
+        setEditedTodoId(null);
+      }
+    } else if (!editedTodoTitle) {
+      handleDeleteEmptyTodo(editedTodoId as number);
+    }
```

Comment:
Avoid `else`/`else if` and nested `if` statements. Try to simplify and make it more readable